homebytwo/importers/utils.py

```diff
@@ -98,3 +107,96 @@ def get_proxy_class_from_data_source(data_source):
         raise Http404("Data Source does not exist")
     else:
         return route_class
+
+
+def download_zip_file(url: str) -> ZipFile:
+    """
+    download zip file from remote url to bytes buffer
+    and wrap it in ZipFile
+    """
+
+    block_size = 1024
+    bytes_io = BytesIO()
+
+    with Session() as session:
+        response = session.get(url, stream=True)
+        response.raise_for_status()
+        file_size = int(response.headers["Content-Length"])
+
+        for data in tqdm(
+            response.iter_content(block_size),
+            total=int(file_size / block_size),
+            unit="B",
+            unit_scale=block_size,
+            desc=f"downloading from {url}",
+        ):
+            bytes_io.write(data)
+
+        return ZipFile(bytes_io)
```

Comment:
@sephii quick question about the approach: is downloading to an in-memory buffer going to kill the server's memory? Should we rather save the file locally. This is all magic to me!

Reply:
Yes, that’s a bad idea, especially `bytes_io.write(data)`, which will store all the downloaded data in memory. You should rather download to a file on disk (eg. with [tempfile](https://docs.python.org/3/library/tempfile.html), just make sure to delete it afterwards (for example by creating your own `ZipFile` class).

Reply:
OK, thanks!